src/autowiring/config_descriptor.h.orig

```diff
@@ -0,0 +1,355 @@
+    metadata_pack(const metadata_pack& rhs) :
+      pack(rhs.pack),
+      list(ptr_vector_from_tuple<const metadata_base>(pack))
+    {}
+
+    metadata_pack(Ms&&... ms) :
+      pack(metadata<Ms>(std::forward<Ms&&>(ms))...),
+      list(ptr_vector_from_tuple<const metadata_base>(pack))
+    {}
+
+    std::tuple<metadata<Ms>...> pack;
+    std::vector<const metadata_base*> list;
+
+    const std::vector<const metadata_base*>& get_list(void) const override { return list; }
+
+    std::unique_ptr<metadata_pack_base> clone(void) const override {
+      return std::unique_ptr<metadata_pack_base>{
+        new metadata_pack{ *this }
+      };
+    }
+
+  private:
+    template<typename U, int... Ns>
+    void bind(const config_field& configField, U& field, index_tuple<Ns...>) {
+      noop(
+        (std::get<Ns>(pack).template bind<U>(configField, field), false)...
+      );
+    }
+
+  public:
+    void bind(const config_field& configField, void* pField) override {
+      bind<T>(
+        configField,
+        *static_cast<T*>(pField),
+        typename make_index_tuple<sizeof...(Ms)>::type{}
+      );
+    }
+  };
+
+  template<typename M>
+  struct metadata_list :
+    std::vector<metadata<M>*>
+  {};
+
+  template<typename... Ms>
+  struct recursive_single_check {};
+
+  template<typename M, typename... Ms>
+  struct recursive_single_check<M, Ms...> :
+    recursive_single_check<Ms...>
+  {
+    static_assert(
+      is_multi<M>::value || !is_any_same<M, Ms...>::value,
+      "Decoration type M was marked as being a singleton decoration, but appeared more than once in an initializer list"
+    );
+  };
+
+  template<typename T>
+  struct default_value_t {
+    default_value_t(T&& value) :
+      value(std::move(value))
+    {}
+
+    T value;
+  };
+
+  template<typename T>
+  default_value_t<typename std::remove_cv<T>::type> default_value(T&& t) {
+    return { std::forward<T>(t) };
+  }
+
+  struct config_field {
+    config_field(void) = default;
+    config_field(const config_field& rhs) { *this = rhs; }
+
+    template<typename T, typename U, typename... Ms>
+    config_field(const char* name, const char* desc, U T::* memptr, Ms&&... ms) :
+      name(name),
+      description(desc),
+      type(auto_id_t<U>()),
+      offset(reinterpret_cast<size_t>(&(static_cast<T*>(nullptr)->*memptr))),
+<<<<<<< HEAD
+      marshaller{ &get_marshaller<U>() },
+      metadata(new metadata_pack<T, Ms...>{ std::forward<Ms>(ms)... })
+    {
+      noop(
+        (HandleMetadata(memptr, ms), false)...
+      );
+      for (const metadata_base* entry : metadata->get_list())
+        m_mdMap[entry->id()].push_back(entry);
+    }
+
+    template<typename T, typename U, typename... Ms>
+    config_field(const char* name, U T::* memptr, Ms&&... ms) :
+      config_field(name, "", memptr, std::forward<Ms>(ms)...)
+=======
+      default_value(std::make_shared<typename std::remove_cv<U>::type>()),
+      marshaller{ &get_marshaller<U>() }
+>>>>>>> origin/develop
```

Comment:
Actually, given the file name suffix of "orig" I suspect that this *file* needs to be removed from the repo.